packages/dds/tree/src/test/utils.ts

```diff
@@ -26,3 +39,110 @@ export class MockDependent extends SimpleObservingDependent {
 		super((token) => this.tokens.push(token), name);
 	}
 }
+
+/**
+ * A test helper class that manages the creation, connection and retrieval of SharedTrees. Instances of this
+ * class are created via {@link create} and satisfy the {@link ITestObjectProvider} interface.
+ */
+export class TestTreeProvider {
+    private static readonly treeId = "TestSharedTree";
+
+    [tree: number]: SharedTree;
+
+    private readonly provider: ITestObjectProvider;
+    private readonly trees: SharedTree[] = [];
+
+    /**
+     * Create a new {@link TestTreeProvider}. The provider can be populated with trees via subsequent calls to
+     * {@link createTree}. If the number of trees used by a test is known ahead of time, consider using
+     * {@link createTrees} instead.
+     */
+    public static async create(): Promise<TestTreeProvider & ITestObjectProvider> {
+        return new TestTreeProvider() as TestTreeProvider & ITestObjectProvider;
+    }
+
+    /**
+     * Create a new {@link TestTreeProvider} with a number of trees pre-initialized.
+     * @param trees - the number of trees to initialize this provider with. This is the same as calling
+     * {@link create} followed by {@link createTree} _trees_ times.
+     *
+     * @example
+     * ```ts
+     * const trees = await TestTreeProvider.createTrees(2);
+     * assert(trees[0].isAttached());
+     * assert(trees[1].isAttached());
+     * await trees.ensureSynchronized();
+     * ```
+     */
+    public static async createTrees(trees: number): Promise<TestTreeProvider & ITestObjectProvider> {
+        const provider = await this.create();
+        for (let i = 0; i < trees; i++) {
+            await provider.createTree();
+        }
+        return provider;
+    }
+
+    /**
+     * Create and initialize a new {@link SharedTree} that is connected to all other trees from this provider.
+     * @returns the tree that was created. For convenience, the tree can also be accessed via `this[i]` where
+     * _i_ is the index of the tree in order of creation.
+     */
+    public async createTree(): Promise<SharedTree> {
+        return this.createTreeAtIndex(this.trees.length);
+    }
+
+    public [Symbol.iterator](): IterableIterator<SharedTree> {
+        return this.trees[Symbol.iterator]();
+    }
+
+    private constructor() {
+        const factory = new SharedTreeFactory();
+        const registry = [[TestTreeProvider.treeId, factory]] as ChannelFactoryRegistry;
+        const driver = new LocalServerTestDriver();
+        this.provider = new TestObjectProvider(
+            Loader,
+            driver,
+            () => new TestContainerRuntimeFactory(
+                "@fluid-example/test-dataStore",
+                new TestFluidObjectFactory(registry),
+            ),
+        );
+
```

Comment:
I personally would stick to idiomatic JS/TS, if possible.

Perhaps expose '.trees' as a Readonly<[]> or add get/set methods?

Reply:
Yea, you are probably right. I was trying to find a way to be able to reference the trees without having to alias them all (trees[0] vs `const tree0 = ...`) like we did in legacy shared tree tests. I suppose `provider.trees[0]` is longer than `trees[0]` but less confusing. And, we may want to add more fields later, like `provider.container[0]`. I think I'll change it.